flexvolume/smb/deployment/smb-flexvol-installer/Dockerfile

```diff
@@ -1,4 +1,33 @@
+###################################################
+FROM python:slim as statically-linked-deps
+###################################################
+
+ARG INSTALL_DEPS=true
+ENV SOURCE_DIR="/install-bin"
+
+RUN \
+  mkdir -p ${SOURCE_DIR} && \
+  apt-get update && \
+  apt-get install --no-install-recommends -y \
+    binutils \
+    cifs-utils \
+    jq \
+    patchelf \
+    && \
+  apt-get clean && \
+  pip3 install --no-cache-dir staticx && \
+  staticx `which mount.cifs` ${SOURCE_DIR}/mount.cifs && \
+  chmod +s ${SOURCE_DIR}/mount.cifs && \
+  staticx `which jq` ${SOURCE_DIR}/jq
+
+###################################################
 FROM busybox
+###################################################
+
+ARG INSTALL_DEPS=false
```

Comment:
what's the default value of `INSTALL_DEPS`? in L5, it's `true`? looks like `INSTALL_DEPS` is only used in `install.sh`

Reply:
Hi @andyzhangx ! I'm not sure what L5 is? 

- INSTALL_DEPS is a new docker build argument I added, it controls whether the statically linked `jq` and `mount.cifs` are copied. 
- I set the default value to `false` (don't install statically linked dependency binaries) to replicate the existing behavior on AKS (uses the system installed jq and mount.cifs in /usr/bin, /usr/sbin), to minimize impact on existing Azure AKS users. Therefore, if you use the Docker image directly following the existing README, it'll work the same as before this PR.
- However, if you deploy using the the new Helm chart, the Helm chart overrides the Docker default ENV value for INSTALL_DEPS to be `INSTALL_DEPS=true`. So if you use Helm, you get statically linked `jq` and `mount.cifs`, but if you already deployed the low-level way, this won't change won't cause a compatibility issue.

Sorry, explaining in a confusing way, let me know if that makes sense?

Reply:
Basically:
- Deploying directly with `kubectl` (current users, users following the existing README): default is an identical deploy to before this PR (full legacy compat)
- Deploying using the Helm chart following the Helm README: default helm deploy includes static `jq` and `mount.cifs` added to the system, which makes this work smoothly on non-AKS systems (should work on AKS too).